### PR TITLE
internal/docs/design.md: add Coding Conventions

### PR DESCRIPTION
**Pull request by jba**

### CODE REVIEW

File: internal/docs/design.md [hunk 1]
```diff
@@ -309,6 +309,17 @@ b, err := blob.NewBucket(d, blob.FeatureUnicodeNames)
...
 on the [mailing list](https://groups.google.com/forum/#!forum/go-cloud).
 
+## Coding Conventions
+
+We try to adhere to commonly accepted Go coding conventions, some of which are
+described on the
+[Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments)
+wiki page. We also adopt the following guidelines:
+* Prefer `map[K]V{}` to `make(map[K]V)`. It's more concise.
+* When writing a loop appending to a slice `s`, prefer `var s []T` to `s :=
```
[the remark this answers is not in the document]
Comment (ijt) [reply internal/docs/design.md]: If we write `m := map[K]V{}` then why don't we also write `s := []T{}` for consistency and brevity?

[the remark this answers is not in the document]
Comment (jba) [reply internal/docs/design.md]: There is effectively no difference between a nil slice and an empty slice: they both have length zero, and you can't do anything with them except append to them. nil is the simpler alternative.

There is a big difference between a nil map and an empty map: you can add things to an empty map, but not a nil one. Here the distinction is not between `var m map[K]V` and `m := map[K]V{}`, but between the latter and `m := make(make[K]V)`.

That said, I guess I don't really care if you want to use `make` for maps. I do care about the slice one, though (and it's in that list of readability comments too).

File: internal/docs/design.md [hunk 2]
```diff
@@ -309,6 +309,17 @@ b, err := blob.NewBucket(d, blob.FeatureUnicodeNames)
 Design discussions regarding enforcing portability are ongoing; we welcome input
 on the [mailing list](https://groups.google.com/forum/#!forum/go-cloud).
 
+## Coding Conventions
+
+We try to adhere to commonly accepted Go coding conventions, some of which are
+described on the
+[Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments)
+wiki page. We also adopt the following guidelines:
+* Prefer `map[K]V{}` to `make(map[K]V)`. It's more concise.
```
[the remark this answers is not in the document]
Comment (jba) [reply internal/docs/design.md]: Switched to `-`, but note the section just above uses `*`.

Not only do I not have an `mdformat` command on my workstation, I couldn't even find it on the web! However, I did find http://brettterpstra.com/projects/mdless, which seems to work fine.

**Comment by jba** [comment]
PTAL

File: internal/docs/design.md [hunk 3]
```diff
@@ -309,6 +309,20 @@ b, err := blob.NewBucket(d, blob.FeatureUnicodeNames)
...
 on the [mailing list](https://groups.google.com/forum/#!forum/go-cloud).
 
+## Coding Conventions
+
+We try to adhere to commonly accepted Go coding conventions, some of which are
+described on the [Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments)
+wiki page. We also adopt the following guidelines:
+- Prefer `map[K]V{}` to `make(map[K]V)`. It's more concise.
+- When writing a loop appending to a slice `s`, prefer `var s []T` plus `append`
+  in the loop to `s :-= make([]T, 0, N)` plus append or `make([]T, N)` plus
```
Comment (zombiezen) [inline internal/docs/design.md]: Typo:

```suggestion
  in the loop to `s := make([]T, 0, N)` plus `append` or `make([]T, N)` plus
```

File: internal/docs/design.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (vangent) [reply internal/docs/design.md]: OK, thanks.

**Comment by jba** [comment]
@clausti @shantuo waiting for your LGTM.

**Comment by jba** [comment]
Good point. Done.

**Pull request merged**